src/prog_models/sim_result.py

```diff
@@ -97,9 +137,24 @@ def pop(self, index: int = -1) -> dict:
         Returns:
             dict: Element Removed
         """
+        warn('pop will be deprecated after version 1.5 of ProgPy. The function will be renamed, popbyindex, and users may begin using it under this name now.',
+            DeprecationWarning, stacklevel=2)
+
         self.times.pop(index)
         return self.data.pop(index)
 
+    def popbyindex(self, index: int = -1) -> dict:
```

Comment:
Should be pop_by_index 

See python naming standards here:https://peps.python.org/pep-0008/#function-and-variable-names